boa_engine/src/bytecompiler/function.rs

```diff
@@ -136,6 +136,9 @@ impl FunctionCompiler {
             // ExecuteAsyncModule ( module ): <https://tc39.es/ecma262/#sec-execute-async-module>
             compiler.emit_opcode(Opcode::CreatePromiseCapability);
 
+            // Note: We set it to one so we don't pop return value when we return.
+            compiler.current_stack_value_count += 1;
```

Comment:
Are these still needed, now that `return_value` in on the `Vm`?

Reply:
> Are these still needed, now that `return_value` in on the `Vm`?

There was a bug that was causing a panic with some weird edge case with array destructuring with yield, this panic was fixed in the latest commit and removed `compiler.current_stack_value_count += 1;` :)